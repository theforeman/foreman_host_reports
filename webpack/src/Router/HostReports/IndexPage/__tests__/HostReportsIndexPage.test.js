import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import HostReportsIndexPage from '../IndexPage';

const fixtures = {
  'render with minimal props': {
    search: '',
    history: { location: { search: '' } },
    hostId: '1',
    fetchAndPush: jest.fn(),
    reloadWithSearch: jest.fn(),
    isLoading: false,
    hasError: false,
    hasData: true,
    itemCount: 1,
    canCreate: true,
    sort: {},
    page: 1,
    perPage: 20,
    reports: [
      {
        id: '1',
        hostId: '1',
        hostName: 'foreman.example.com',
        proxyId: '1',
        proxyName: 'foreman.example.com',
        format: 'plain',
        reportedAt: '2021-01-19T12:34:02.841645028Z',
        change: 0,
        nochange: 0,
        failure: 0,
      },
    ],
  },
};

describe('HostReportsIndexPage', () => {
  describe('redering', () =>
    testComponentSnapshotsWithFixtures(HostReportsIndexPage, fixtures));
});
