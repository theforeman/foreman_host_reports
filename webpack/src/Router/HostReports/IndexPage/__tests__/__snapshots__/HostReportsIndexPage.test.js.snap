// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HostReportsIndexPage redering render with minimal props 1`] = `
<PageLayout
  header="Host Reports"
  isLoading={false}
  onBookmarkClick={[MockFunction]}
  onSearch={[MockFunction]}
  searchProps={
    Object {
      "autocomplete": Object {
        "id": "searchBar",
        "searchQuery": "",
        "url": "host_reports/auto_complete_search",
        "useKeyShortcuts": true,
      },
      "bookmarks": Object {
        "canCreateBookmarks": true,
        "documentationUrl": "4.1.5Searching",
        "url": "/api/bookmarks",
      },
      "controller": "host_reports",
    }
  }
  searchQuery=""
  searchable={true}
  toolbarButtons={
    <ExportButton
      title="Export"
      url="/api/v2/host_reports/export"
    />
  }
>
  <WrappedHostReportsTable
    fetchAndPush={[MockFunction]}
    hostId="1"
    itemCount={1}
    pagination={
      Object {
        "page": 1,
        "perPage": 20,
      }
    }
    reloadWithSearch={[MockFunction]}
    results={
      Array [
        Object {
          "change": 0,
          "failure": 0,
          "format": "plain",
          "hostId": "1",
          "hostName": "foreman.example.com",
          "id": "1",
          "nochange": 0,
          "proxyId": "1",
          "proxyName": "foreman.example.com",
          "reportedAt": "2021-01-19T12:34:02.841645028Z",
        },
      ]
    }
    setToDelete={[Function]}
    sort={Object {}}
    toDelete={Object {}}
  />
</PageLayout>
`;
